src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Repositories/DelegationMetadataRepository.cs

```diff
@@ -133,16 +135,83 @@ public async Task<List<DelegationChange>> GetAllDelegationChanges(string altinnA
+                }
+            }
+
+            return delegationChanges;
+        }
+
+        private static DelegationChange GetDelegationChange(NpgsqlDataReader reader)
+        {
+            DelegationChange delegationChange = new DelegationChange();
+            delegationChange.PolicyChangeId = reader.GetValue<int>("delegationchangeid");
+            delegationChange.AltinnAppId = reader.GetValue<string>("altinnappid");
+            delegationChange.OfferedByPartyId = reader.GetValue<int>("offeredbypartyid");
+            delegationChange.CoveredByPartyId = reader.GetValue<int>("coveredbypartyid");
+            delegationChange.CoveredByUserId = reader.GetValue<int>("coveredbyuserid");
+            delegationChange.PerformedByUserId = reader.GetValue<int>("performedbyuserid");
+            delegationChange.BlobStoragePolicyPath = reader.GetValue<string>("blobstoragepolicypath");
+            delegationChange.BlobStorageVersionId = reader.GetValue<string>("blobstorageversionid");
+            delegationChange.IsDeleted = reader.GetValue<bool>("isdeleted");
+            delegationChange.Created = reader.GetValue<DateTime>("created");
+            return delegationChange;
+        }
+
+        private async Task<List<DelegationChange>> GetAllCurrentDelegationChangesPartyIds(List<string> altinnAppIds = null, List<int> offeredByPartyIds = null, List<int> coveredByPartyIds = null)
         {
             try
             {
                 using NpgsqlConnection conn = new NpgsqlConnection(_connectionString);
                 await conn.OpenAsync();
 
-                NpgsqlCommand pgcom = new NpgsqlCommand(getAllCurrentDelegationChangesSql, conn);
+                NpgsqlCommand pgcom = new NpgsqlCommand(getAllCurrentDelegationChangesPartyIdsSql, conn);
                 pgcom.Parameters.AddWithValue("_altinnAppIds", NpgsqlTypes.NpgsqlDbType.Array | NpgsqlTypes.NpgsqlDbType.Text, altinnAppIds?.Count > 0 ? altinnAppIds : DBNull.Value);
                 pgcom.Parameters.AddWithValue("_offeredByPartyIds", NpgsqlTypes.NpgsqlDbType.Array | NpgsqlTypes.NpgsqlDbType.Integer, offeredByPartyIds?.Count > 0 ? offeredByPartyIds : DBNull.Value);
                 pgcom.Parameters.AddWithValue("_coveredByPartyIds", NpgsqlTypes.NpgsqlDbType.Array | NpgsqlTypes.NpgsqlDbType.Integer, coveredByPartyIds?.Count > 0 ? coveredByPartyIds : DBNull.Value);
+
+                List<DelegationChange> delegationChanges = new List<DelegationChange>();
+
+                using NpgsqlDataReader reader = pgcom.ExecuteReader();
+                while (reader.Read())
+                {
+                    delegationChanges.Add(GetDelegationChange(reader));
+                }
+
+                return delegationChanges;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Authorization // PostgresRepository // GetAllDelegationChanges // Exception");
```

Comment:
all the _logger calls should be updated:
PostgresRepository -> DelegationMetadataRepository
GetAllDelegationChanges -> actual operation name

Reply:
Updated these